Dockerfile

```diff
@@ -1,94 +1,88 @@
 FROM snapcore/snapcraft AS snapcraft
 # Using multi-stage dockerfile to obtain snapcraft binary
 
+FROM ubuntu:bionic AS flutterbuilder
+RUN apt-get update \
+    && apt-get install -y \
+        git curl unzip
+# Install Flutter from the beta channel
+RUN git clone --single-branch --depth=1 --branch beta https://github.com/flutter/flutter /opt/flutter 2>&1 \
+    && /opt/flutter/bin/flutter doctor -v
+
 FROM ubuntu:bionic AS xarbuilder
 RUN apt-get update \
 	&& apt-get install -y \
-		wget tar libssl1.0-dev libxml2-dev make g++
+		git libssl1.0-dev libxml2-dev make g++ autoconf
 RUN cd /tmp \
-	&& wget https://github.com/downloads/mackyle/xar/xar-1.6.1.tar.gz 2>&1 \
-	&& tar -zxvf xar-1.6.1.tar.gz 2>&1 \
-	&& mv xar-1.6.1 xar \
-	&& cd xar \
+    && git clone https://github.com/mackyle/xar 2>&1 \
```

Comment:
I would prefer to checkout the tag `xar-1.6.1` to avoid unexpected breaks when changes are pushed to xar's master branch.

Reply:
xar isn't develop anymore, but I see what you mean